//  File: linux_admin_rpc.proto
//  Author: Tom Roeder <tmroeder@google.com>
//
//  Description: Protocol buffers for LinuxAdminRPCPC.
//
//  Copyright (c) 2013, Google Inc.  All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package tao;

enum LinuxAdminRPCRequestType {
  LINUX_ADMIN_RPC_UNKNOWN = 0;  // unused
  LINUX_ADMIN_RPC_SHUTDOWN = 1;
  LINUX_ADMIN_RPC_START_HOSTED_PROGRAM = 2;
  LINUX_ADMIN_RPC_STOP_HOSTED_PROGRAM = 3;
  LINUX_ADMIN_RPC_KILL_HOSTED_PROGRAM = 4;
  LINUX_ADMIN_RPC_GET_TAO_HOST_NAME = 5;
  LINUX_ADMIN_RPC_LIST_HOSTED_PROGRAMS = 6;
}

message LinuxAdminRPCRequest {
  required LinuxAdminRPCRequestType rpc = 1;
  required uint64 seq = 2;
  optional bytes data = 3;
  optional string path = 4;
  repeated string args = 5;
}

message LinuxAdminRPCHostedProgramList {
  repeated string name = 1;
  repeated int32 pid = 2;
}

message LinuxAdminRPCResponse {
  required LinuxAdminRPCRequestType rpc = 1;
  required uint64 seq = 2;
  optional string error = 3;
  optional bytes data = 4;    // either a string or
                              // serialized LinuxAdminRPCHostedProgramList
  optional string reason = 5; // reason in case of failure
}

